src/enclave/main.cpp

```diff
@@ -104,10 +104,10 @@ extern "C"
       {
         std::lock_guard<SpinLock> guard(create_lock);
 
-        tid = enclave::ThreadMessaging::thread_count.fetch_add(1);
-        num_pending_threads.fetch_sub(1);
-        thread_ids.emplace(std::pair<std::thread::id, uint16_t>(
+        tid = threading::ThreadMessaging::thread_count.fetch_add(1);
+        threading::thread_ids.emplace(std::pair<std::thread::id, uint16_t>(
           std::this_thread::get_id(), tid));
+        num_pending_threads.fetch_sub(1);
```

Comment:
This may just be paranoia, but I think this ordering is safer.

If we decrement `num_pending_threads` _first_, then we allow other threads to advance past the busy-wait below before this thread has been added to the map. When we emplace, we're potentially moving the entries other threads are trying to read.

By putting ourselves in the map and _then_ decrementing `num_pending_threads`, we provide a strict gating: the map is fully populated before any thread can proceed.